lib/test.js

```diff
@@ -53,6 +53,77 @@ function plan(count) {
 	this.plan(count, captureStack(this.plan));
 }
 
+function tryTest(fn) {
+	const args = [];
+	for (let i = 1, len = arguments.length; i < len; i++) {
+		args.push(arguments[i]);
+	}
+
+	let commitCalled = false;
+	let discarded = false;
+	let finished = false;
+
+	const commitFn = (ret, toCommit) => {
```

Comment:
Could you remove the `toCommit` argument? It's odd to pass `false` in order to get discard behavior. That should be a separate function.

Reply:
I added this so that there is no duplicate code between `commit()` and `discard()` functions returned to the user. The user dont have access to call this function directly, it is proxied. I will remove it though, for clarity between `commit()` and `discard()` functions.